enhancements/cluster-logging/forwarding-json-structured-logs.md

```diff
@@ -128,154 +104,96 @@ New *pipeline* field:
 For each log entry; if `parse: json` is set _and_ the entry is valid JSON, the output record will include a `structured` field _equivalent_ to the JSON entry.
 It may differ in field order and use of white-space.
 
-All record will have a `message` field for backwards compatibility.
+### Output type elasticsearch
 
-#### Output  configuration for type:elasticsearch
+For most output types it is sufficient to enable `parse: json` to forward JSON data.
+Elasticsearch is a special case; JSON records with _different formats_ must go to different indices, otherwise type conflicts and cardinality problems can occur.
+The elasticsearch output can be configured with a "structured type" that is used to construct an index name.
 
-New fields for output type `elasticsearch`
-* `indexKey`: (string, optional) Use value of meta-data key as `index` value, if present.
-  These keys are supported:
-  - `kubernetes.namespaceName`: Use the namespace name as the index name.
-  - `kubernetes.labels.<key>`: Use the string value of kubernetes label with key `<key>`.
-  - `openshift.labels.<key>`: Use the string value of an openshift label with key `<key>` (see [](forwarder-tagging.md))
-  - *other keys may be added in future*
-* `indexName`: (string, optional)
-  * If `indexKey` is not set, or the key is missing, use `indexName` as the `index` value.
-  * If `indexKey` is set and the key is present, that takes precedence over `indexName`
+New fields in  `output.elasticsearch`:
+* `structuredTypeName`: (string, optional) the structured type, unless `structuredTypeKey` is set, and the key is present.
+* `structuredTypeKey`: (string, optional) Use the value of this meta-data key (if present and non-empty)  as the structured type. These keys are supported:
+  * `kubernetes.labels.<key>`: Use the string value of kubernetes label with key `<key>`.
+  * `openshift.labels.<key>`: Use the string value of an openshift label with key `<key>` (see [forwarder-tagging.md]())
 
-**Note:** The Elasticsearch output will _delete_ the `structured` field if the rules above do not yield a non-empty index name.
-This is to avoid sending randomly structured JSON to shared indices.
+Notes:
+* The Elasticsearch _index_ for structured records is formed by appending "app-" to the structured type.
```

Comment:
s/appending/prepending/

Reply:
we likely want to ensure `-write` is appended though

Reply:
Done